src/api.js

```diff
@@ -0,0 +1,797 @@
+- Create a statement that contains parameters like '$VVV', ':VVV', '@VVV'
+- Call Statement.bind with an object as parameter
+
+## Binding values to parameters
+@example Bind values to anonymous parameters
+    var stmt = db.prepare("UPDATE test SET a=? WHERE id BETWEEN ? AND ?");
+    stmt.bind([5, 10, 20]);
+ - Create a statement that contains parameters like '?', '?NNN'
+ - Call Statement.bind with an array as parameter
+
+## Value types
+Javascript type | SQLite type
+--- | ---
+number | REAL, INTEGER
+boolean | INTEGER
+string | TEXT
+Array, Uint8Array | BLOB
+null | NULL
+@see http://www.sqlite.org/datatype3.html
+
+@see http://www.sqlite.org/lang_expr.html#varparam
+@param values [Array,Object] The values to bind
+@return [Boolean] true if it worked
+@throw [String] SQLite Error
+ */
+Statement.prototype["bind"] = function(values) {
+    if (!this.stmt) {
+        throw "Statement closed";
+    }
+    this["reset"]();
+    if (Array.isArray(values)) {
+        return this.bindFromArray(values);
+    } else {
+        return this.bindFromObject(values);
+    }
+};
+
+/* Execute the statement, fetching the the next line of result,
+that can be retrieved with [Statement.get()](#get-dynamic) .
+
+@return [Boolean] true if a row of result available
+@throw [String] SQLite Error
+ */
+Statement.prototype["step"] = function() {
+    var ret;
+    if (!this.stmt) {
+        throw "Statement closed";
+    }
+    this.pos = 1;
+    switch (ret = sqlite3_step(this.stmt)) {
+    case SQLITE_ROW:
+        return true;
+    case SQLITE_DONE:
+        return false;
+    default:
+        return this.db.handleError(ret);
+    }
+};
+
+Statement.prototype.getNumber = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_double(this.stmt, pos);
+};
+
+Statement.prototype.getString = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_text(this.stmt, pos);
+};
+
+Statement.prototype.getBlob = function(pos) {
+    var i, k, ptr, ref, result, size;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    size = sqlite3_column_bytes(this.stmt, pos);
+    ptr = sqlite3_column_blob(this.stmt, pos);
+    result = new Uint8Array(size);
+    for (i = k = 0, ref = size; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
+        result[i] = HEAP8[ptr + i];
+    }
+    return result;
+};
+
+/* Get one row of results of a statement.
+If the first parameter is not provided, step must have been called before get.
+@param [Array,Object] Optional: If set, the values will be bound to the statement, and it will be executed
+@return [Array<String,Number,Uint8Array,null>] One row of result
+
+@example Print all the rows of the table test to the console
+
+    var stmt = db.prepare("SELECT * FROM test");
+    while (stmt.step()) console.log(stmt.get());
+ */
+Statement.prototype["get"] = function(params) {
+    var field, k, ref, results1;
```

Comment:
the `var` could be directly in front of the variables

Reply:
applied suggestion in commit 43191e7a8453b83d7bd73d77754d41f1cbaf8730